.github/copilot-instructions.md

```diff
@@ -0,0 +1,316 @@
+  - `frontend/src/main.ts` - Angular Application bootstrap entry point
+  - `frontend/src/react` - React components (currently only used for experimental BlockNote integration)
+  - `frontend/src/stimulus` - Stimulus controllers, helpers
+  - `frontend/src/turbo` - Turbo integration (e.g. custom Turbo Stream actions)
+- `lib/` - Ruby libraries and extensions
+- `lookbook/` - Lookbook component previews for ViewComponents (see https://github.com/lookbook-hq/lookbook)
+- `modules/` - OpenProject plugin modules
+- `spec/` - RSpec test suite
+  - `spec/features/` - System/feature tests (Capybara)
+  - `spec/models/` - Model unit tests
+  - `spec/requests/` - API/integration tests
+  - `spec/services/` - Service tests
+
+### Configuration Files
+- `.erb_lint.yml` - ERB template linting
+- `.rubocop.yml` - Ruby linting rules
+- `.ruby-version` - Ruby version (check this file for current version)
+- `docker-compose.yml` - Docker development environment
+- `frontend/eslint.config.mjs` - JavaScript/TypeScript linting
+- `Gemfile` / `Gemfile.lock` - Ruby dependencies
+- `lefthook.yml` - Git hooks configuration
+- `package.json` / `frontend/package.json` - Node.js dependencies
+- `Procfile.dev` - Services for `bin/dev`
+
+## GitHub Actions CI/CD
+
+### Main Workflows
+- **test-core.yml** - Main test suite (units + features, ~40 min, runs on all PRs)
+- **rubocop-core.yml** - Ruby linting (runs on all PRs with Ruby changes)
+- **eslint-core.yml** - JS/TS linting (runs on all PRs with JS/TS changes)
+- **test-frontend-unit.yml** - Frontend unit tests
+- **brakeman-scan-core.yml** - Security scanning
+- **codeql-scan-core.yml** - Code quality/security analysis
+
+### CI Requirements for Merge
+- All linting checks must pass (Rubocop, ESLint, erb_lint)
+- Test suite must be green
+- No security vulnerabilities introduced (Brakeman, CodeQL)
+
+**Skip CI**: Add `[skip ci]` to commit message to skip CI (use sparingly).
+
+## Common Issues and Workarounds
+
+### Database Configuration
+- **Issue**: Docker fails with "database.yml exists"
+- **Fix**: Delete or rename `config/database.yml` when using Docker
+
+### Memory Issues in Docker
+- **Issue**: Frontend container exits with status 137
+- **Fix**: Increase Docker memory limit to at least 4GB
+
+### Test Failures on CI but Passing Locally
+- Run with `CI=true` environment variable (eager loads app)
+- Check for `OPENPROJECT_*` environment variables
+- Match the random seed: `bundle exec rspec --seed 18352`
+- Use `--bisect` to find order-dependent failures
+- View browser tests with `OPENPROJECT_TESTING_NO_HEADLESS=1`
+
+### Frontend Build Issues
+- **Issue**: "jQuery not defined", frontend asset errors, or blank page
+- **Fix**: Run `bin/setup_dev` to rebuild frontend completely
+
+### Parallel Test Failures
+- Tests run in parallel on CI with different random seeds per group
+- Check `tmp/parallel_runtime.log` for execution times
+
```

Comment:
Mention Flaky specs as a possible issue. Reference `docs/development/testing/handling-flaky-tests/README.md`

https://github.com/opf/openproject/commit/0108374bd30efbea63de1e179644a0f45437ec22

cc / @cbliard

Reply:
@cbliard this is just FYI. no need to do anything for now!